src/v/wasm/wasi.cc

```diff
@@ -79,7 +83,16 @@ uint32_t log_writer::flush() {
         amt += b.size();
     }
     auto level = _is_guest_stdout ? ss::log_level::info : ss::log_level::warn;
-    _logger->log(level, "{} - {}", _name, fmt::join(buf, ""));
+    auto joined = absl::StrJoin(buf, "");
+    if (!is_valid_utf8(joined) || contains_control_character(joined))
+      [[unlikely]] {
+        joined = absl::CHexEscape(joined);
+    }
+    constexpr static size_t max_log_line = 2_KiB;
+    if (joined.size() > max_log_line) {
+        joined.resize(max_log_line);
```

Comment:
Is there any reason to be concerned about weird behavior if we cut off the log line in the middle of some escape sequence?

Reply:
Good question - I don't believe so.